includes/class-wc-admin-loader.php

```diff
@@ -0,0 +1,533 @@
+		return " $admin_body_class ";
+	}
+
+	/**
+	 * Removes notices that should not be displayed on WC Admin pages.
+	 *
+	 * @todo move to activity panel code?
+	 */
+	public static function remove_notices() {
+		if ( ! self::is_admin_page() && ! self::is_embed_page() ) {
+			return;
+		}
+
+		// Hello Dolly.
+		if ( function_exists( 'hello_dolly' ) ) {
+			remove_action( 'admin_notices', 'hello_dolly' );
+		}
+	}
+
+	/**
+	 * Runs before admin notices action and hides them.
+	 *
+	 * @todo move to activity panel code?
+	 */
+	public static function inject_before_notices() {
+		if ( ( ! self::is_admin_page() && ! self::is_embed_page() ) || ! self::is_feature_enabled( 'activity-panels' ) ) {
+			return;
+		}
+		echo '<div class="woocommerce-layout__notice-list-hide" id="wp__notice-list">';
+		echo '<div class="wp-header-end" id="woocommerce-layout__notice-catcher"></div>'; // https://github.com/WordPress/WordPress/blob/f6a37e7d39e2534d05b9e542045174498edfe536/wp-admin/js/common.js#L737.
+	}
+
+	/**
+	 * Runs after admin notices and closes div.
+	 *
+	 * @todo move to activity panel code?
+	 */
+	public static function inject_after_notices() {
+		if ( ( ! self::is_admin_page() && ! self::is_embed_page() ) || ! self::is_feature_enabled( 'activity-panels' ) ) {
+			return;
+		}
+		echo '</div>';
+	}
+
+	/**
+	 * Edits Admin title based on section of wc-admin.
+	 *
+	 * @param string $admin_title Modifies admin title.
+	 * @todo Can we do some URL rewriting so we can figure out which page they are on server side?
+	 */
+	public static function update_admin_title( $admin_title ) {
+		if ( ! self::is_admin_page() && ! self::is_embed_page() ) {
+			return $admin_title;
+		}
+
+		if ( self::is_embed_page() ) {
+			$sections = wc_admin_get_embed_breadcrumbs(); // @todo Bring this function into loader.
+			$sections = is_array( $sections ) ? $sections : array( $sections );
+			$pieces   = array();
+
+			foreach ( $sections as $section ) {
+				$pieces[] = is_array( $section ) ? $section[1] : $section;
+			}
+
+			$pieces = array_reverse( $pieces );
+			$title  = implode( ' &lsaquo; ', $pieces );
+		} else {
+			$title = __( 'Dashboard', 'woocommerce-admin' );
+		}
+		/* translators: %1$s: updated title, %2$s: blog info name */
+		return sprintf( __( '%1$s &lsaquo; %2$s &#8212; WooCommerce', 'woocommerce-admin' ), $title, get_bloginfo( 'name' ) );
+	}
+
+	/**
+	 * Set up a div for the app to render into.
+	 */
+	public static function page_wrapper() {
+		?>
+		<div class="wrap">
+			<div id="root"></div>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Hooks extra neccessary data into the component settings array already set in WooCommerce core.
+	 *
+	 * @param array $settings Array of component settings.
+	 * @return array Array of component settings.
+	 */
+	public static function add_component_settings( $settings ) {
+		// @todo Some of these component settings should be added by feature specific code.
+		$preload_data_endpoints = array(
```

Comment:
Maybe this could use a filter, so the other components you mention can just hook into the filter?

Reply:
Good idea. 

Added a filter here https://github.com/woocommerce/woocommerce-admin/pull/1863/commits/3d8ab59c461dbe2f9131d68dc40bd82280b9e219#diff-1ae4b2e500c5ac085f2c35ba19fc730aR450, and hooked in from the specific features: https://github.com/woocommerce/woocommerce-admin/pull/1863/commits/3d8ab59c461dbe2f9131d68dc40bd82280b9e219#diff-1c906050672400ebd97de57489183b8cR44